src/backend/linalg_internal_cpu/Det_internal.cpp

```diff
@@ -27,8 +31,8 @@ namespace cytnx {
         "[ERROR][Det_internal] Error in Lapack function 'zgetrf': Lapack INFO = ", info);
       od[0] = 1;
       bool neg = 0;
-      for (lapack_int i = 0; i < N; i++) {
-        od[0] *= (cytnx_complex128)_Rin[i * N + i];
+      for (int i = 0; i < N; i++) {
+        od[0] *= cytnx_complex128(_Rin[i * N + i]);
```

Comment:
please make sure this casting will work if the element is not complex

Reply:
Whether `lapack_complex_double` is defined as `std::complex<double>`(c++ complex) or `double _Complex`(C complex) depends on whether MKL is used.  The spec of C++17 doesn't promise to construct `std::complex<double>` from `double _Complex`, but it is allowed in the implementation of GCC and clang.

Reply:
A safe solution is updated.